src/AzureIoTHub.Portal/Server/Helpers/DeviceHelper.cs

```diff
@@ -87,13 +87,13 @@ public static string RetrieveDesiredPropertyValue(Twin twin, string propertyName
         /// Set the desired property value.
         /// </summary>
         /// <param name="twin">Device twin.</param>
-        /// <param name="propertyName">Property to set.</param>
+        /// <param name="propertyRowKey">Row key of the property to set.</param>
         /// <param name="value">Property value.</param>
-        public static void SetDesiredProperty(Twin twin, string propertyName, object value)
+        public static void SetDesiredProperty(Twin twin, string propertyRowKey, object value)
         {
             ArgumentNullException.ThrowIfNull(twin, nameof(twin));
 
-            twin.Properties.Desired[propertyName] = value;
+            twin.Properties.Desired[propertyRowKey] = value;
```

Comment:
Should use name here not the rown key, since the rown key is sanitized ?

Reply:
I rollback this method. But we always have to link devices and properties using the field property RowKey because it is sanitized.

In the other hand, if we link devices and properties using the field name, it will produces errors because the field property name is not sanitized.